feed/feed-inbound/feed-inbound-1.0.0/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/RssInject.java

```diff
@@ -0,0 +1,137 @@
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.UUIDGenerator;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.builder.Builder;
+import org.apache.axis2.builder.BuilderUtil;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.transport.TransportUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.mediators.base.SequenceMediator;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+/**
+ * RssInject uses to mediate the received Feeds
+ */
+public class RssInject implements InjectHandler {
+    private static final Log log = LogFactory.getLog(RssInject.class);
+    private final String injectingSeq;
+    private final String onErrorSeq;
+    private final boolean sequential;
+    private final SynapseEnvironment synapseEnvironment;
+    private final String contentType;
+
+    public RssInject(String injectingSeq, String onErrorSeq, boolean sequential,
+                     SynapseEnvironment synapseEnvironment, String contentType) {
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.sequential = sequential;
+        this.synapseEnvironment = synapseEnvironment;
+        this.contentType = contentType;
+    }
+
+    /**
+     * Determine the message builder to use, set the feed message to the
+     * message context and
+     * inject the message to the sequence
+     */
+    public boolean invoke(Object object) {
+        org.apache.synapse.MessageContext msgCtx = null;
+        try {
+            msgCtx = createMessageContext();
+        } catch (Exception e) {
+            log.error(e.getMessage());
+        }
+        MessageContext axis2MsgCtx = null;
+        try {
+            if (msgCtx != null) {
+                axis2MsgCtx =
+                        ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx).getAxis2MessageContext();
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        // Determine the message builder to use
+        Builder builder = null;
+        try {
+            if (axis2MsgCtx != null) {
+                builder = BuilderUtil.getBuilderFromSelector(contentType, axis2MsgCtx);
+            }
+        } catch (AxisFault axisFault) {
+            log.error("Error while creating message builder :: " + axisFault.getMessage(),
+                    axisFault);
+        }
+        if (builder == null) {
+            if (log.isDebugEnabled()) {
+                log.info("No message builder found for type '" + contentType +
+                        "'. Falling back to SOAP.");
+            }
+        }
+        try {
+            OMElement documentElement = (OMElement) object;
+            assert msgCtx != null;
```

Comment:
Assertion statement is mainly used for testing purpose. You can't use here please use the "if" condition.